src/modeling_library/mixture.jl

```diff
@@ -1,33 +1,47 @@
+
 ##################################################################
 # homogeneous mixture: arbitrary number of the same distribution #
 ##################################################################
 
 """
     @hom_mixture output_type base_dist arg_dims
 
-Defines a new type for a mixture distribution family, and evaluates to a singleton element of that type.
+Defines a new type for a mixture distribution family, and evaluates to a
+singleton element of that type.
 
-The first argument, `output_type`, should match the output type of the base distribution.
+The first argument, `output_type`, should match the output type of the base
+distribution.
 
-The second argument `base_dist` must have type Gen.Distribution, and defines the distribution family of each component in the mixture.
+The second argument `base_dist` must have type Gen.Distribution, and defines
+the distribution family of each component in the mixture.
 
-The third argument `arg_dims` must be a literal tuple of Ints, of length equal to the number of arguments taken by the base distribution family.
-A value of 0 at a position in the tuple an indicates that the corresponding argument to the base distribution is a scalar, and integer values of i for i >= 1 indicate that the corresponding argument is an i-dimensional array.
+The third argument `arg_dims` must be a literal tuple of Ints, of length equal
+to the number of arguments taken by the base distribution family.  A value of 0
+at a position in the tuple an indicates that the corresponding argument to the
+base distribution is a scalar, and integer values of i for i >= 1 indicate that
+the corresponding argument is an i-dimensional array.
 
 Example:
 
     mixture_of_normals = @hom_mixture Float64 normal (0, 0)
 """
```

Comment:
Can we update these docs to also describe the usage of the resulting mixture distribution? (It's not obvious to me that the calling pattern is `dist(weights, arg1_for_all_components, arg2_for_all_components, ...)`.)

Reply:
> * Why do we need this to be a macro, rather than a 'distribution combinator'? Is it that this keeps more avenues open for performance engineering in the future?
> *

That was ostensibly the idea. But I'll see what a non-macro version looks like.